web/src/features/fba/components/map/FBAMap.tsx

```diff
@@ -120,38 +141,55 @@ const FBAMap = (props: FBAMapProps) => {
       if (layer) {
         map.removeLayer(layer)
       }
-      map.addLayer(fireZoneVector)
+      map.addLayer(fireZoneVTL)
+      map.addLayer(fireZoneLabelVTL)
+      map.on('click', event => {
+        fireZoneVTL.getFeatures(event.pixel).then(features => {
+          if (!features.length) {
+            props.setSelectedFireZoneID(null)
+            return
+          }
+          const feature = features[0]
+          if (!feature) {
+            return
+          }
+          const zoneExtent = feature.getGeometry()?.getExtent()
+          if (!isUndefined(zoneExtent)) {
+            map.getView().fit(zoneExtent)
+          }
+          props.setSelectedFireZoneID(feature.get('mof_fire_zone_id'))
+        })
+      })
     }
-  }, [map, fireZoneVector])
+  }, [map]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    fetchFireZoneAreas(props.runType, props.runDate.toISO(), props.forDate.toISODate())
```

Comment:
The current pattern is the page will do all the fetching calls to keep FBAMap isolated with its own local state but I can adjust this in the advisory color PR